components/dashboard/src/projects/Project.tsx

```diff
@@ -159,40 +137,39 @@ export default function () {
                     </ItemField>
                 </Item>
                 {branches.map((branch, index) => {
+                    if (!filter(branch)) {
+                        return undefined;
+                    }
```

Comment:
Nit: Wouldn't it be cleaner to filter out branches before the `.map`? E.g. like so:

```
{branches.filter(filter).map((branch, index) => {
    // ...
```

Reply:
sounds good! will apply in the following PR.